### PR TITLE
HCMPRE-585 - Added save plan facility persister config

### PR DESCRIPTION
**Pull request by Taniya-eGov**
## Summary by CodeRabbit

- **New Features**
	- Introduced functionality for persisting plan facility data.
	- Added a new version entry (1.0.0) to enhance data management.
	- Implemented a transaction for saving plan facility details.
	- Included a new SQL query for inserting data into the plan facility linkage.

### CODE REVIEW

**Comment by Taniya-eGov** [comment]
@coderabbitai review

File: health/egov-persister/plan-service-persister.yml [hunk 1]
```diff
@@ -494,3 +494,42 @@ serviceMaps:
...
+
+            - jsonPath: $.PlanFacility.tenantId
+
+            - jsonPath: $.PlanFacility.planConfigurationId
+
+            - jsonPath: $.PlanFacility.facilityId
+
+            - jsonPath: $.PlanFacility.residingBoundary
+
+            - jsonPath: $.PlanFacility.serviceBoundaries
```
Comment (Priyanka-eGov) [inline health/egov-persister/plan-service-persister.yml:517]: why is serviceBoundaries of type JSONB, in the db design its of type TEXT

File: health/egov-persister/plan-service-persister.yml [hunk 2]
```diff
@@ -494,3 +494,42 @@ serviceMaps:
...
 
             - jsonPath: $.Plan.auditDetails.lastModifiedTime
+
+    - version: 1.0.0
+      description: Persists the plan facility
+      fromTopic: save-plan-facility
+      isTransaction: true
+      isAuditEnabled: true
+      queryMaps:
+        - query: INSERT INTO plan_facility_linkage(id, tenant_id, plan_configuration_id, facility_id, residing_boundary, service_boundaries, additional_details, active ,created_by, created_time, last_modified_by, last_modified_time) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?,?);
```
Comment (Priyanka-eGov) [inline health/egov-persister/plan-service-persister.yml:504]: ?,?); in consistent spacing
 , active ,

**Review by Priyanka-eGov — CHANGES_REQUESTED** [review]

File: health/egov-persister/plan-service-persister.yml (hunk 1 above)
Comment (Taniya-eGov) [reply health/egov-persister/plan-service-persister.yml:517]: Updated to type TEXT for consistency with db

File: health/egov-persister/plan-service-persister.yml (hunk 2 above)
Comment (Taniya-eGov) [reply health/egov-persister/plan-service-persister.yml:504]: Made correct spacing

**Pull request merged**